src/github.com/getlantern/flashlight/client/balancer.go

```diff
@@ -14,9 +15,18 @@ func (client *Client) initBalancer(cfg *ClientConfig) *balancer.Balancer {
 	dialers := make([]*balancer.Dialer, 0, len(cfg.FrontedServers)+len(cfg.ChainedServers))
 
 	log.Debugf("Adding %d domain fronted servers", len(cfg.FrontedServers))
+	var highestQOSFrontedDialer fronted.Dialer
+	highestQOS := -1
 	for _, s := range cfg.FrontedServers {
-		dialer := s.dialer(cfg.MasqueradeSets)
+		fd, dialer := s.dialer(cfg.MasqueradeSets)
 		dialers = append(dialers, dialer)
+		if dialer.QOS > highestQOS {
+			highestQOSFrontedDialer = fd
+		}
+	}
+
+	if highestQOSFrontedDialer != nil {
+		go determinePublicIp(highestQOSFrontedDialer)
 	}
```

Comment:
Cool thanks. That's certainly true for this iteration, but if/when we add
back in existing functionality that public IP becomes crucial for
determining if you're in a censored region.

On Wed, Feb 18, 2015 at 12:07 PM, oxtoacart <notifications@github.com>
wrote:

> In src/github.com/getlantern/flashlight/client/balancer.go
> <https://github.com/getlantern/flashlight-build/pull/59#discussion_r24935700>
> :
>
> >  		dialers = append(dialers, dialer)
> > +		if dialer.QOS > highestQOS {
> > +			highestQOSFrontedDialer = fd
> > +		}
> > +	}
> > +
> > +	if highestQOSFrontedDialer != nil {
> > +		go determinePublicIp(highestQOSFrontedDialer)
> >  	}
>
> Done. I made it debug since it doesn't actually prevent the functioning of
> the system, it just keeps us from getting a public ip.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/getlantern/flashlight-build/pull/59/files#r24935700>.
>



-- 
--
President
Brave New Software Project, Inc.
https://www.getlantern.org
pgp A998 2B6E EF1C 373E 723F A813 045D A255 901A FD89